MediaBrowser.Api/QuickConnect/QuickConnectService.cs

```diff
@@ -0,0 +1,132 @@
+        public string FriendlyName { get; set; }
+    }
+
+    [Route("/QuickConnect/Connect", "GET", Summary = "Attempts to retrieve authentication information")]
+    public class Connect : IReturn<QuickConnectResult>
+    {
+        [ApiMember(Name = "Secret", Description = "Quick connect secret", IsRequired = true, DataType = "string", ParameterType = "query", Verb = "GET")]
+        public string Secret { get; set; }
+    }
+
+    [Route("/QuickConnect/Authorize", "POST", Summary = "Authorizes a pending quick connect request")]
+    [Authenticated]
+    public class Authorize : IReturn<bool>
+    {
+        [ApiMember(Name = "Code", Description = "Quick connect identifying code", IsRequired = true, DataType = "string", ParameterType = "query", Verb = "GET")]
+        public string Code { get; set; }
+    }
+
+    [Route("/QuickConnect/Deauthorize", "POST", Summary = "Deletes all quick connect authorization tokens for the current user")]
+    [Authenticated]
+    public class Deauthorize : IReturn<int>
+    {
+        [ApiMember(Name = "UserId", Description = "User Id", IsRequired = false, DataType = "string", ParameterType = "query", Verb = "GET")]
+        public Guid UserId { get; set; }
+    }
+
+    [Route("/QuickConnect/Status", "GET", Summary = "Gets the current quick connect state")]
+    public class QuickConnectStatus : IReturn<QuickConnectResult>
+    {
+
+    }
+
+    [Route("/QuickConnect/Available", "POST", Summary = "Enables or disables quick connect")]
+    [Authenticated(Roles = "Admin")]
+    public class Available : IReturn<QuickConnectState>
+    {
+        [ApiMember(Name = "Status", Description = "New quick connect status", IsRequired = false, DataType = "QuickConnectState", ParameterType = "query", Verb = "GET")]
+        public QuickConnectState Status { get; set; }
+    }
+
+    [Route("/QuickConnect/Activate", "POST", Summary = "Temporarily activates quick connect for the time period defined in the server configuration")]
+    [Authenticated]
+    public class Activate : IReturn<bool>
+    {
+
+    }
+
+    public class QuickConnectService : BaseApiService
+    {
+        private IQuickConnect _quickConnect;
+        private IUserManager _userManager;
+        private IAuthorizationContext _authContext;
+
+        public QuickConnectService(
+            ILogger<QuickConnectService> logger,
+            IServerConfigurationManager serverConfigurationManager,
+            IHttpResultFactory httpResultFactory,
+            IUserManager userManager,
+            IAuthorizationContext authContext,
+            IQuickConnect quickConnect)
+            : base(logger, serverConfigurationManager, httpResultFactory)
+        {
+            _userManager = userManager;
+            _quickConnect = quickConnect;
+            _authContext = authContext;
+        }
+
+        public object Get(Initiate request)
+        {
+            return _quickConnect.TryConnect(request.FriendlyName);
```

Comment:
Currently this will always return a `200`. I think the response should be changed depending on Success / Fail.